Source/WebKit/WebProcess/WebPage/FindController.cpp

```diff
@@ -208,21 +205,21 @@ void FindController::updateFindUIAfterPageScroll(bool found, const String& strin
             if (m_foundStringMatchIndex >= (int) matchCount)
                 m_foundStringMatchIndex -= matchCount;
         }
+    }
 
-        // If updating UI after finding an individual match, update the current
-        // match rects and inform the UI process that we succeeded.
-        // If we're doing a multi-result search and just updating the indicator,
-        // this would blow away the results for the other matches.
-        // FIXME: This whole class needs a much clearer division between these two paths.
-        if (originator == FindUIOriginator::FindString) {
-            m_findMatches.clear();
-            Vector<IntRect> matchRects;
-            if (auto range = m_webPage->corePage()->selection().firstRange()) {
-                matchRects = RenderObject::absoluteTextRects(*range);
-                m_findMatches.append(*range);
-            }
-            m_webPage->send(Messages::WebPageProxy::DidFindString(string, matchRects, matchCount, m_foundStringMatchIndex, didWrap == DidWrap::Yes));
+    // If updating UI after finding an individual match, update the current
+    // match rects and inform the UI process that we succeeded.
+    // If we're doing a multi-result search and just updating the indicator,
+    // this would blow away the results for the other matches.
+    // FIXME: This whole class needs a much clearer division between these two paths.
+    if (completionHandler) {
+        m_findMatches.clear();
+        Vector<IntRect> matchRects;
+        if (auto range = m_webPage->corePage()->selection().firstRange()) {
+            matchRects = RenderObject::absoluteTextRects(*range);
+            m_findMatches.append(*range);
         }
+        completionHandler(idOfFrameContainingString, matchRects, matchCount, m_foundStringMatchIndex, didWrap == DidWrap::Yes);
```

Comment:
This is an unnecessary Vector copy.  Make the parameter a Vector<IntRect>&& instead of Vector<IntRect> and you'll avoid these.